piksi_tools/console/console.py

```diff
@@ -888,15 +887,28 @@ class PortChooser(HasTraits):
         width=400,
         title='Swift Console - Select Piksi Interface', )
 
-    def __init__(self, baudrate=None):
+    def refresh_ports(self):
+        """
+        This method refreshes the port list
+        """
         try:
-            self.ports = [p for p, _, _ in s.get_ports()]
-            if baudrate not in BAUD_LIST:
-                self.choose_baud = False
-            self.baudrate = baudrate
+          self.ports = [p for p, _, _ in s.get_ports()]
         except TypeError:
             pass
 
+    def __init__(self, baudrate=None):
+        self.refresh_ports()
+        # As default value, use the first city in the list:
+        try:
+            self.port = self.ports[0]
+        except IndexError:
+            pass
+        if baudrate not in BAUD_LIST:
+            self.choose_baud = False
+        self.baudrate = baudrate
+        self.update_ports = call_repeatedly(0.5, self.refresh_ports)
```

Comment:
@denniszollo Seems like this isn't letting the process ever exit cleanly on mac os when quiting the selector.